### PR TITLE
add referral text if a user is referred

### PR DESCRIPTION
**Pull request by amaliwanag2**

### CODE REVIEW

File: src/pages/cats.js [hunk 1]
```diff
@@ -177,10 +177,27 @@ const useStyles = makeStyles(() => ({
...
+    if (userReferrerId !== null && userReferrerId !== undefined) {
+      isReferred = true
+    }
+    return isReferred
+  }
   const [
     showUnsupportedBrowserMessage,
     setShowUnsupportedBrowserMessage,
   ] = useState(false)
+  const [isReferral] = useState(checkIfReferrered())
```
Comment (kmjennison) [inline src/pages/cats.js:200]: I believe it's best practice to set client-determined state after mount. Here, we'd start with a default state of `false`. Gatsby is rendering page HTML on build (always no referrer), and if the state changes before mount (client side with `u` param), there will be a hydration conflict.

File: src/pages/cats.js [hunk 2]
```diff
@@ -343,6 +360,20 @@ const Cats = ({ pageContext }) => {
...
+                  data-test-id={'referral-text'}
+                  style={{
+                    width: '80%',
+                    textAlign: 'justify',
+                    fontWeight: 'bold',
+                  }}
+                >
+                  Since you were referred to join Tab for Cats, we'll give an
+                  additional donation to shelter cats on behalf of you and your
+                  referral when you sign up!
```
Comment (kmjennison) [inline src/pages/cats.js:374]: Maybe: "Your friend sent you a gift / By signing up with this link, you'll immediately help shelter cats get adopted by giving them 5 treats for positive-reinforcement training."

**Review by kmjennison — CHANGES_REQUESTED** [review]

File: src/pages/cats.js [hunk 3]
```diff
@@ -206,6 +213,7 @@ const Cats = ({ pageContext }) => {
         STORAGE_REFERRAL_DATA_REFERRING_USER,
         userReferrerId
       )
+      setIsReferral(true)
```
Comment (amaliwanag2) [inline src/pages/cats.js:216]: @kmjennison now just only showing based on url

File: src/pages/__tests__/cats.test.js [hunk 4]
```diff
@@ -93,6 +104,19 @@ describe('cats page', () => {
     expect(localStorageMgr.setItem).not.toHaveBeenCalled()
   })
 
+  it('does not shows referral copy when referrer id is included as a URL parameter', () => {
```
Comment (kmjennison) [inline src/pages/__tests__/cats.test.js:107]: Fix test description

File: src/pages/__tests__/cats.test.js [hunk 5]
```diff
@@ -142,6 +183,18 @@ describe('cats page', () => {
     expect(localStorageMgr.setItem).not.toHaveBeenCalled()
   })
 
+  it('does not show referral copy the referrer ID is not in the URL params', () => {
```
Comment (kmjennison) [inline src/pages/__tests__/cats.test.js:186]: This is the same test as above? Delete.

File: src/pages/cats.js [hunk 6]
```diff
@@ -260,67 +268,84 @@ const Cats = ({ pageContext }) => {
...
-            <div style={{ margin: 0 }}>
-              <div>
-                <Star style={{ color: '#ffc533', width: 18, height: 18 }} />
-                <Star style={{ color: '#ffc533', width: 18, height: 18 }} />
-                <Star style={{ color: '#ffc533', width: 18, height: 18 }} />
-                <Star style={{ color: '#ffc533', width: 18, height: 18 }} />
-                <StarHalf style={{ color: '#ffc533', width: 18, height: 18 }} />
-              </div>
+        <div className={cx.SectionHeight}>
+          {isReferral ? (
```
Comment (kmjennison) [inline src/pages/cats.js:272]: Design is awkward. Make it full width and either pin to the top (so it doesn't reflow content on mount) or below the main content on the first page?

Screen Shot 2021-04-13 at 4 47 29 PM

**Review by kmjennison — CHANGES_REQUESTED** [review]

File: src/pages/cats.js (hunk 6 above)
Comment (amaliwanag2) [reply src/pages/cats.js:278]: kept some side margin but pinned to top.  Felt like the best look

**Review by kmjennison — APPROVED** [review]
LGTM 🍝

**Pull request merged**